pkg/persistence/postgrespersister.go

```diff
@@ -205,17 +256,36 @@ func (p *PostgresPersister) retrieveVersionFromTable(tableName string) (*string,
 func (p *PostgresPersister) saveVersionToTable(tableName string, versionNumber *string) error {
 	dbVersionStruct := postgres.Version{
 		Version:           versionNumber,
-		ServiceName:       crawlerServiceName,
-		LastUpdatedDateTs: ctime.CurrentEpochSecsInInt64()}
-
-	queryString := cpostgres.InsertIntoDBQueryString(tableName, postgres.Version{})
+		ServiceName:       CrawlerServiceName,
+		LastUpdatedDateTs: ctime.CurrentEpochSecsInInt64(),
+		Exists:            true}
+	onConflict := fmt.Sprintf("%s, %s", postgres.VersionFieldName, postgres.ServiceFieldName)
+	updateFields := []string{postgres.LastUpdatedTsFieldName, postgres.ExistsFieldName}
+	queryString := p.upsertVersionDataQueryString(tableName, dbVersionStruct, onConflict,
+		updateFields)
 	_, err := p.db.NamedExec(queryString, dbVersionStruct)
 	if err != nil {
 		return fmt.Errorf("Error saving version to table: %v", err)
 	}
 	return nil
 }
 
+func (p *PostgresPersister) upsertVersionDataQueryString(tableName string, dbModelStruct interface{},
+	onConflict string, updatedFields []string) string {
+	var queryBuf bytes.Buffer
+	fieldNames, fieldNamesColon := cpostgres.StructFieldsForQuery(dbModelStruct, true, "")
+	queryString := fmt.Sprintf("INSERT INTO %s (%s) VALUES(%s) ON CONFLICT(%s) DO UPDATE SET ",
+		tableName, fieldNames, fieldNamesColon, onConflict) // nolint: gosec
+	queryBuf.WriteString(queryString) // nolint: gosec
+	for idx, field := range updatedFields {
+		queryBuf.WriteString(fmt.Sprintf("%s=:%s", field, field)) // nolint: gosec
+		if idx+1 < len(updatedFields) {
```

Comment:
this is fine, but you could also use `strings.Join` which seems cleaner.

Reply:
Oh yea strings.join uses the internal string builder